lib/libfigpar/figpar.c

```diff
@@ -268,19 +266,18 @@ parse_config(struct figpar_config options[], const char *path,
 			 */
 
 			/* Get the current offset */
-			charpos = lseek(fd, 0, SEEK_CUR) - 1;
-			if (charpos == -1) {
+			if ((charpos = lseek(fd, 0, SEEK_CUR)) == -1) {
 				close(fd);
 				return (-1);
 			}
+			charpos--;
 
 			/*
 			 * Go back so we can read the character before the key
 			 * to check if the character is escaped (which means we
 			 * should continue).
 			 */
-			error = (int)lseek(fd, -2, SEEK_CUR);
-			if (error == -3) {
+			if (lseek(fd, -2, SEEK_CUR) == -1) {
```

Comment:
The original code is a bit odd here...  -3 would be Linux-speak for ESRCH (a define that goes back to V7 times, so it's the same everywhere). But there's no ESRCH documented anywhere I can find for lseek (in linux or elsewhere).

Reply:
ESRCH means no process found matching the process ID, so for lseek() this check seems was wrong.

Reply:
I spoke with Devin privately, and she mentioned that the use of -3 wasn't actually intended to represent ESRCH as she wrote figpar before ESRCH existed.

Reply:
ESRCH (3) is an errno value, however it's checking lseek()'s return value.

Reply:
ESRCH has been around since at least V3 Unix (even though userland errno didn't seem to exist until V4... with the caveat that our extant artifacts are limited beyond the man pages).
But it's clear we should check -1, so I'm landing it...